### PR TITLE
Add timestamp support to sequence function

### PR DESCRIPTION
**Pull request by kewang1024**
Added
 - [x] sequence(timestamp, timestamp, interval day to second)

Followup
 - [ ] [Optimization] Add fast path when step is constant
 - [ ] sequence(date, date, interval year to month)
 - [ ] sequence(timestamp, timestamp, interval year to month)

### CODE REVIEW

**Comment by kewang1024** [comment]
To keep consistent with the current java presto worker behavior, we also calculate timestamp up to millisecond and neglect the data afterwards (microsecond and nanosecond) for sequence function

File: velox/functions/prestosql/Sequence.cpp [hunk 1]
```diff
@@ -131,20 +141,44 @@ class SequenceFunction : public exec::VectorFunction {
...
+      int64_t stop,
+      DecodedVector* stepVector,
+      vector_size_t row,
+      bool isDate) {
+    if (stepVector) {
+      auto step = stepVector->valueAt<int64_t>(row);
+      if (isDate) {
+        VELOX_USER_CHECK(
+            step % kMillisInDay == 0,
+            "sequence step must be a day interval if start and end values are dates");
```
Comment (spershin) [inline velox/functions/prestosql/Sequence.cpp:174]: Should we perform this check only in checkArguments() so we only have that in the 1st loop and not the 2nd?

**Comment by spershin** [comment]
> To keep consistent with the current java presto worker behavior, we also calculate timestamp up to millisecond and neglect the data afterwards (microsecond and nanosecond) for sequence function

@kewang1024 

That answers one of my questions.
Let's add this as comment in the code where we converts timestamp to ms.

File: velox/functions/prestosql/Sequence.cpp [hunk 2]
```diff
@@ -35,6 +35,11 @@ int64_t toInt64(Date value) {
   return value.days();
 }
 
+template <>
+int64_t toInt64(Timestamp value) {
```
Comment (laithsakka) [inline velox/functions/prestosql/Sequence.cpp:39]: const Timestamp& value,

File: velox/functions/prestosql/Sequence.cpp [hunk 3]
```diff
@@ -48,6 +53,11 @@ Date add(Date value, int64_t steps) {
   return Date(value.days() + steps);
 }
 
+template <>
+Timestamp add(Timestamp value, int64_t steps) {
```
Comment (laithsakka) [inline velox/functions/prestosql/Sequence.cpp:57]: ditto

File: velox/functions/prestosql/Sequence.cpp [hunk 4]
```diff
@@ -60,6 +70,7 @@ class SequenceFunction : public exec::VectorFunction {
       const TypePtr& outputType,
       exec::EvalCtx& context,
       VectorPtr& result) const override {
+    bool isDate = args[0]->type() == DATE();
```
Comment (laithsakka) [inline velox/functions/prestosql/Sequence.cpp:73]: shall we check kind. is Data() constructing shared Ptr

**Review by laithsakka — APPROVED** [review]
Code looks good

File: velox/functions/prestosql/Sequence.cpp (hunk 2 above)
Comment (kewang1024) [reply velox/functions/prestosql/Sequence.cpp:39]: Unfortunately, If this is changed, the template parameter wouldn't match

**Pull request closed without merging**